src/Helpers.cc

```diff
@@ -76,39 +76,40 @@ namespace ignition
       return outputString.str();
     }
 
-    // The following regex takes a time string in the general format of
-    // "dd hh:mm:ss.nnn" where n is milliseconds, if just one number is
-    // provided, it is assumed to be seconds
-    static const std::regex time_regex(
-        "^([0-9]+ ){0,1}"                       // day:
-                                                // Any positive integer
-
-        "(?:([1-9]:|[0-1][0-9]:|2[0-3]:){0,1}"  // hour:
-                                                // 1 - 9:
-                                                // 01 - 19:
-                                                // 20 - 23:
-
-        "([0-9]:|[0-5][0-9]:)){0,1}"            // minute:
-                                                // 0 - 9:
-                                                // 00 - 59:
-
-        "(?:([0-9]|[0-5][0-9]){0,1}"            // second:
-                                                // 0 - 9
-                                                // 00 - 59
-
-        "(\\.[0-9]{1,3}){0,1})$");              // millisecond:
-                                                // .0 - .9
-                                                // .00 - .99
-                                                // .000 - 0.999
-
-
     /////////////////////////////////////////////
     bool splitTimeBasedOnTimeRegex(
         const std::string &_timeString,
         uint64_t & numberDays, uint64_t & numberHours,
         uint64_t & numberMinutes, uint64_t & numberSeconds,
         uint64_t & numberMilliseconds)
     {
+      // The following regex takes a time string in the general format of
+      // "dd hh:mm:ss.nnn" where n is milliseconds, if just one number is
+      // provided, it is assumed to be seconds.  We construct upon first use
+      // and never destroy it, in order to avoid the [Static Initialization
+      // Order Fiasco](https://en.cppreference.com/w/cpp/language/siof).
+      static const std::regex * const time_regex = new std::regex(
```

Comment:
I should probably rename this to kTimeRegex, to comply with the Google Style Guide?